gcloud/bigquery/job.py

```diff
@@ -634,6 +660,30 @@ def _scrub_local_properties(self, cleaned):
         schema = cleaned.pop('schema', {'fields': ()})
         self.schema = _parse_schema_resource(schema)
 
+    @classmethod
+    def from_api_repr(cls, resource, client):
+        """Factory:  construct a job given its API representation
+
+        :type resource: dict
+        :param resource: dataset job representation returned from the API
+
+        :type client: :class:`gcloud.bigquery.client.Client`
+        :param client: Client which holds credentials and project
+                       configuration for the dataset.
+
+        :rtype: :class:`gcloud.bigquery.job.LoadTableFromStorageJob`
+        :returns: Job parsed from ``resource``.
+        """
+        name, config = cls._get_resource_config(resource)
+        dest_config = config['destinationTable']
+        assert dest_config['projectId'] == client.project
```

Comment:
I think we need a better strategy than "raise ValueError" for cases where the back-end returns unexpected data.  The [Python library docs](https://docs.python.org/2/library/exceptions.html) describe ValueError as:

> Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.

In these cases, there is no issue with values passed *into* a function or method: the issue is with data fetched from a back-end source.